R/getStepwiseDivergence.R

```diff
@@ -222,3 +233,36 @@ setMethod("getTimeDivergence",
     return(x)
 
 }
+
+############################ HELPER FUNCTION(S) ################################
+
+.get_mat_from_se <- function (x, assay_name, altexp) {
+    # non-empty string
+    if(!.is_non_empty_string(assay_name)){
+        stop("'assay_name' must be a non-empty single character value.",
+             call. = FALSE)
+    }
+    if (!is.null(altexp)) {
+        x <- altExp(x, altexp)
+    }
+    mat <- NULL
+    if (assay_name %in% assayNames(x)) {
+        mat <- t(assay(x, assay_name))
```

Comment:
just wondering is it really necessary to transpose (and convert the matrix into unexpected way)

Reply:
intention was not to break the existing flow (if I understood right the task), since afterwards rows are supposed to be samples. And assay call to the resulting x or altExp(x) would produce features as rows.

Reply:
In general, in TreeSE assays, rows are features and columns are samples.

Reply:
Yes, but when mat is returned from the function then line 212 assumes mat having samples on rows.

Reply:
as long as there are no unnecessary transposes it is fine

Reply:
in principle, it would be most clear if the function that should pick SE assay, would return it in the same way than it is contained in the actual assay.

And then if vegan functions need transpose, it could be transposed during that function call.